modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/adapters/BaseNuxeoArtifactDocAdapter.java

```diff
@@ -88,6 +88,15 @@ public DocumentModel getDoc() {
         return doc;
     }
 
+    /**
+     * Updates the document behind this adapter.
+     *
+     * @since 11.2
+     */
+    public void setDoc(DocumentModel doc) {
+        this.doc = doc;
+    }
```

Comment:
I would have said that document adapter is immutable regarding their doc.

From what I understand this is used in a sub class, therefore we could make it protected to only allow this usage, wdyt?

Reply:
This update might not be needed anyway, i'll check